third_party/move/move-compiler-v2/transactional-tests/tests/no-v1-comparison/assert_one.move

```diff
@@ -0,0 +1,34 @@
+//# publish
+module 0x42::Test {
+    struct Tester has drop {
+        f: u64
+    }
+
+    public fun new(): Tester {
+        Tester { f: 10 }
+    }
+
+    public fun len(t: &Tester): u64 {
+        t.f
+    }
+
+    public fun modify(t: &mut Tester): u64 {
+        t.f = t.f - 1;
+        9 - t.f
+    }
+}
+
+//# run
+script {
+use 0x42::Test;
+fun main() {
+    let x = Test::new();
+    assert!(Test::len(&x) == 10, 70002);
+
+    let i = 0;
+    while (i < Test::len(&x)) {
+        assert!(Test::modify(&mut x) == i + 1);  // this will fail
```

Comment:
So currently, this behavior (of allowing 1 arg to `assert!`) is compiler v2 only? In that case, I think we should make this a Move 2 language change. If not, depending on the compiler used, we can have differing behavior for Move 1, without any warning or indication.

Reply:
It is a Move 2 feature. Lets add a test to checking-lang-v1 directory to verify we get the right error message for v1

Reply:
Done.